cpg-language-java/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/java/DeclarationHandler.kt

```diff
@@ -198,112 +198,42 @@ open class DeclarationHandler(lang: JavaLanguageFrontend) :
-                    }
-                    else -> {
-                        log.debug(
-                            "Member {} of type {} is something that we do not parse yet: {}",
-                            decl,
-                            recordDeclaration.name,
-                            decl.javaClass.simpleName
-                        )
-                    }
-                }
-        }
-        if (recordDeclaration.constructors.isEmpty()) {
-            val constructorDeclaration =
-                this.newConstructorDeclaration(
-                        recordDeclaration.name.localName,
-                        recordDeclaration,
-                    )
-                    .implicit(recordDeclaration.name.localName)
-            recordDeclaration.addConstructor(constructorDeclaration)
-            frontend.scopeManager.addDeclaration(constructorDeclaration)
-        }
-        frontend.processAnnotations(recordDeclaration, classInterDecl)
+        handleRecordMembers(classInterDecl, recordDeclaration)
         frontend.scopeManager.leaveScope(recordDeclaration)
 
-        // We need special handling if this is a so called "inner class". In this case we need to
-        // store
-        // a "this" reference to the outer class, so methods can use a "qualified this"
-        // (OuterClass.this.someFunction()). This is the same as the java compiler does. The
-        // reference
-        // is stored as an implicit field.
         if (frontend.scopeManager.currentScope is RecordScope) {
-            // Get all the information of the outer class (its name and the respective type). We
-            // need this to generate the field.
-            val scope = frontend.scopeManager.currentScope as RecordScope?
-            if (scope?.name != null) {
-                val fieldType = scope.name?.let { this.objectType(it) } ?: unknownType()
-
-                // Enter the scope of the inner class because the new field belongs there.
-                frontend.scopeManager.enterScope(recordDeclaration)
-                val field =
-                    this.newFieldDeclaration("this$" + scope.name?.localName, fieldType, listOf())
-                        .implicit("this$" + scope.name?.localName)
-                frontend.scopeManager.addDeclaration(field)
-                frontend.scopeManager.leaveScope(recordDeclaration)
-            }
+            // We need special handling if this is a so called "inner class". In this case we need
+            // to
+            // store
+            // a "this" reference to the outer class, so methods can use a "qualified this"
+            // (OuterClass.this.someFunction()). This is the same as the java compiler does. The
+            // reference
+            // is stored as an implicit field.
+            handleInnerRecord(recordDeclaration)
         }
         return recordDeclaration
     }
 
+    private fun handleInnerRecord(recordDeclaration: RecordDeclaration) {
```

Comment:
Here, we should avoid naming the function `handler` as that is our convention for functions that get an AST object from a parser and return an AST object in our CPG representation. Handlers are also added to a handler map in the init block. Here, I would ask for a descriptive name that does not collide with `handler`, e.g. resolve, process, add, etc.